manifests/init.pp

```diff
@@ -61,6 +61,13 @@
   Hash[String, Hash] $pools6                              = {},
 ) inherits dhcp::params {
 
+  # check if extra_config is a string, if so convert it to an array
+  if $extra_config =~ String {
```

Comment:
I think this needs an additional check so the empty string is filtered out. Users might have that in their profile classes and ``` <% if @extra_config && !@extra_config.empty? -%>``` will be true for ```['']```

Reply:
What about:

```puppet
Optional[Variant[Array[String[1]],String[1]]] $extra_config
```

So valid are Strings with at least one char or arrays with strings with at least one char.

Reply:
IMHO that would make it a breaking change although I do like it (besides the ```Optional```).

Reply:
It would be a breaking change. But I don't see a usecase for anybody explicitly setting it to an empty string.  Do you think we can do 
```
Variant[Array[String[1]],String[1]]
```

and release it as a minor update? (IMO yes)

Reply:
I think that's fair and it's not like there's undefined behavior. It's a clean compilation error that's easy to fix. 👍